src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -253,15 +252,18 @@ private static List<ISegment> GetBlockOfLines(DiffLineInfo[] diffLines, DiffLine
         return result;
     }
 
-    private static List<TextMarker> GetDifferenceMarkers(Func<ISegment, string> getText, IReadOnlyList<ISegment> linesRemoved, IReadOnlyList<ISegment> linesAdded, int beginOffset)
+    private static IEnumerable<TextMarker> GetDifferenceMarkers(string text, IReadOnlyList<ISegment> linesRemoved, IReadOnlyList<ISegment> linesAdded, int beginOffset)
     {
         List<TextMarker> markers = [];
-        foreach ((ISegment lineRemoved, ISegment lineAdded) in LinesMatcher.FindLinePairs(getText, linesRemoved, linesAdded))
+        foreach ((ISegment lineRemoved, ISegment lineAdded) in LinesMatcher.FindLinePairs(GetText, linesRemoved, linesAdded))
         {
-            AddDifferenceMarkers(markers, getText, lineRemoved, lineAdded, beginOffset);
+            AddDifferenceMarkers(markers, GetText, lineRemoved, lineAdded, beginOffset);
         }
 
         return markers;
+
+        string GetText(ISegment line)
+            => text.Substring(line.Offset + beginOffset, line.Length - beginOffset);
```

Comment:
```suggestion
            => text[line.Offset + beginOffset .. line.Offset + line.Length];
```

Reply:
spread not possible in a function

Reply:
OK, range indexer needs parens:
```suggestion
            => text[(line.Offset + beginOffset)..(line.Offset + line.Length)];
```